### PR TITLE
Add reversible syntax for change_column_default

### PR DESCRIPTION
**Pull request by sikachu**
Passing `:from` and `:to` to `change_column_default` makes this command
reversible as user has defined its previous state.

So, instead of having the migration command as:

```
change_column_default(:posts, :state, "draft")
```

They can write it as:

```
change_column_default(:posts, :state, from: nil, to: "draft")
```

This also includes a change to guide to reflect this.

### CODE REVIEW

File: guides/source/active_record_migrations.md [hunk 1]
```diff
@@ -436,8 +436,17 @@ change_column_default :products, :approved, false
...
-is reversible.
+Additionaly, for `change_column_default`, you can pass a Hash which defines the
+changes. This will make the command reversible. The command above can be
+rewritten as:
+
+```ruby
+change_column_default :products, :approved, from: true, to: false
+```
+
+TIP: `change_column` command is always irreversible, `change_column_null` is
```
Comment (egilburg) [inline guides/source/active_record_migrations.md]: IMO this tip is redundant especially with so many clauses, should be enough to just clearly document reversibility of each command in that command's description.

File: guides/source/active_record_migrations.md [hunk 2]
```diff
@@ -436,8 +436,17 @@ change_column_default :products, :approved, false
 This sets `:name` field on products to a `NOT NULL` column and the default
 value of the `:approved` field to false.
 
-TIP: Unlike `change_column` (and `change_column_default`), `change_column_null`
-is reversible.
+Additionaly, for `change_column_default`, you can pass a Hash which defines the
```
Comment (egilburg) [inline guides/source/active_record_migrations.md]: We should encourage writing reversible migrations by default. Let's change default documentation to use `from:` and `to:`, and here write something like:
 
```Note: You could also write the above migration as `change_column_default :products, :approved, false`, but unlike the previous example, this would make your migration irreversible.```

File: guides/source/active_record_migrations.md [hunk 3]
```diff
@@ -436,8 +436,17 @@ change_column_default :products, :approved, false
...
+changes. This will make the command reversible. The command above can be
+rewritten as:
+
+```ruby
+change_column_default :products, :approved, from: true, to: false
+```
+
+TIP: `change_column` command is always irreversible, `change_column_null` is
+always reversible, and `change_column_default` is only reversible if using
```
Comment (egilburg) [inline guides/source/active_record_migrations.md]: Not in scope of this PR, but would be nice if we could extract keys from `change_column` command, and normalize them to reversible migrations as long as all keys are reversible. E.g. combining `null` and `default` often goes together in a migration that sets null to false:

```ruby
change_column 'foo', null: false, default: { from: nil, to: 'bar' }
```

Would be interpreted as:

```ruby
change_column_default 'foo', from: nil, to: 'bar'
change_column_null 'foo', false
```

Above order is important, if null is disallowed the default must be changed first, if null is allowed the default must change last.

File: guides/source/active_record_migrations.md (hunk 2 above)
Comment (sikachu) [inline guides/source/active_record_migrations.md]: Oh, I like that. Let me update the doc.

File: activerecord/lib/active_record/migration/command_recorder.rb [hunk 4]
```diff
@@ -164,6 +164,16 @@ def invert_remove_index(args)
       alias :invert_add_belongs_to :invert_add_reference
       alias :invert_remove_belongs_to :invert_remove_reference
 
+      def invert_change_column_default(args)
+        table, column, options = *args
+
+        unless options && options.is_a?(Hash) && options[:from] && options[:to]
```
Comment (carlosantoniodasilva) [inline activerecord/lib/active_record/migration/command_recorder.rb]: Wouldn't this fail if from or to are nil/false?

Comment (sikachu) [inline activerecord/lib/active_record/migration/command_recorder.rb]: Good question. I would expect that it should, but [the test](https://github.com/rails/rails/pull/20018/files#diff-9531a83a96f7d0e6d2611e8494f50c96R172) didn't fail. Let me look further into this.

Comment (carlosantoniodasilva) [inline activerecord/lib/active_record/migration/command_recorder.rb]: That might be because the test uses strings, try adding one more to test booleans just in case :)

File: guides/source/active_record_migrations.md (hunk 3 above)
Comment (sikachu) [inline guides/source/active_record_migrations.md]: Noted. I'm going to submit another PR for it.

File: activerecord/lib/active_record/migration/command_recorder.rb (hunk 4 above)
Comment (sikachu) [inline activerecord/lib/active_record/migration/command_recorder.rb]: Yep, it failed indeed. Good catch.

I'm going to push an update to this.

**Comment by sikachu** [comment]
@egilburg @carlosantoniodasilva would you mind give this PR another look and maybe merge it? Thanks!

**Comment by rafaelfranca** [comment]
Awesome! Thank you @sikachu
